src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,33 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+public abstract class Contact {
+	
+	public final String value;
+	private boolean isPrivate;
+	
+	public Contact(String contact, boolean isPrivate, String validationRegex, String messageConstraints) throws IllegalValueException {
+		this.value = contact;
+		this.isPrivate = isPrivate;
+		if(!isValidContact(contact, validationRegex)) {
+				throw new IllegalValueException(messageConstraints);
```

Comment:
Indentation problem here?. Our coding standard requires you to use 4 spaces instead of tabs. You can configure Eclipse to convert tabs to spaces automatically.